examples/mip/bilinear.py

```diff
@@ -0,0 +1,28 @@
+# Inspired by https://www.gurobi.com/documentation/9.5/examples/bilinear_py.html#subsubsection:bilinear.py
+#
+# This example formulates and solves the following simple bilinear model:
+#  maximize    x
+#  subject to  x + y + z <= 10
+#              x * y <= 2         (bilinear inequality)
+#              x * z + y * z = 1  (bilinear equality)
+#              x, y, z non-negative (x integral in second version)
+
+from symfit import parameters, Eq
+from symfit import MIP
+
+# Create variables
+x, y, z = parameters('x, y, z')
+
+objective = 1.0 * x
+constraints = [
+    x + y + z <= 10,
+    x*y <= 2,
+    Eq(x*z + y*z, 1),
+]
+
+mip = MIP(- objective, constraints=constraints)
+# For full control, we still need to access the model directly.
+mip.backend.model.Params.NonConvex = 2
```

Comment:
Oh I *don't* like this at all...

Reply:
Nope, it's horrible. The good news is that it is possible however, which means that we don't have to wrap everything even to get started. But we still need to think about a syntax to deal with this. Personally I think that it might be best if the backend class GurobiBackend accepts any options, which will then be passed on to gurobi without looking into it.

The example above would then become
```python
backend = GurobiBackend(NonConvex=2)
mip = MIP(- objective, constraints=constraints, backend=backend)

```